deeplay/components/dcgan/dcgan.py

```diff
@@ -0,0 +1,464 @@
+from typing import List, Optional, Literal, Any, Sequence, Type, overload, Union
+
+from ... import (
+    DeeplayModule,
+    Layer,
+    LayerList,
+    Sequential,
+    LayerActivation,
+    LayerActivationNormalization,
+)
+
+import torch
+import torch.nn as nn
+
+from deeplay import LayerList
+
+
+class ConvBlock(DeeplayModule):
+    """
+    Convolutional block for DCGAN discriminator. It consists of a 2D Convolutional layer, a Batch Normalization layer and a LeakyReLU activation function.
+    """
+
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        kernel_size,
+        stride,
+        padding,
+        batch_norm=True,
+    ):
+        super().__init__()
+        self.blocks = LayerList()
+        self.blocks.append(
+            LayerActivationNormalization(
+                Layer(
+                    nn.Conv2d,
+                    in_channels,
+                    out_channels,
+                    kernel_size,
+                    stride,
+                    padding,
+                    bias=not batch_norm,
+                ),
+                Layer(nn.BatchNorm2d, out_channels)
+                if batch_norm
+                else Layer(nn.Identity),
+                Layer(nn.LeakyReLU, 0.2),
+            )
+        )
+
+    def forward(self, x):
+        for block in self.blocks:
+            x = block(x)
+        return x
+
+
+class DcganDiscriminator(DeeplayModule):
```

Comment:
Since it's a exact architecture, we should move to something like models/dcgan/discriminator.py

We should also decide if we use the suggestion by Jesus to make it a function

Reply:
Yeah, I'm holding on the file relocations until we decide on it. 

For the second comment, for fixed architectures I'm not sure if it is useful to make it as function. Since the blocks would be specific to the fixed architectures, it would be good to leave them standalone if people want to check how it is built.